### PR TITLE
docs(Hooks): Add each Hook to a title

### PR DESCRIPTION
**Pull request by RafaelGSS**
#### Checklist

Improve: #1593  

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by RafaelGSS** [comment]
Maybe add more comments for each hook sounds good

File: docs/Hooks.md [hunk 1]
```diff
@@ -236,9 +168,57 @@ fastify.addHook('onSend', (request, reply, payload, done) => {
...
+  if (err) {
+    throw new Error('some errors occurred.')
+  }
+  return
+})
+```
+
 The `onResponse` hook is executed when a response has been sent, so you will not be able to send more data to the client, however you can use this hook to send some data to an external service or elaborate some statistics.
 
+## Request/Reply Hooks
```
Comment (Eomm) [inline docs/Hooks.md]: I would move this title after the TOC

File: docs/Hooks.md [hunk 2]
```diff
@@ -236,9 +168,57 @@ fastify.addHook('onSend', (request, reply, payload, done) => {
...
+
+[Request](https://github.com/fastify/fastify/blob/master/docs/Request.md) and [Reply](https://github.com/fastify/fastify/blob/master/docs/Reply.md) are the core Fastify objects.<br/>
+`done` is the function to continue with the [lifecycle](https://github.com/fastify/fastify/blob/master/docs/Lifecycle.md).
+
+It is pretty easy to understand where each hook is executed by looking at the [lifecycle page](https://github.com/fastify/fastify/blob/master/docs/Lifecycle.md).<br>
```
Comment (Eomm) [inline docs/Hooks.md]: I would move this line in the main paragraph, in the `# Hooks` title

File: docs/Hooks.md [hunk 3]
```diff
@@ -236,9 +168,57 @@ fastify.addHook('onSend', (request, reply, payload, done) => {
...
+Hooks are affected by Fastify's encapsulation, and can thus be applied to selected routes. See the [Scopes](#scope) section for more information.
+
```
Comment (Eomm) [inline docs/Hooks.md]:
```suggestion

### Manage Errors from a hook
```

File: docs/Hooks.md [hunk 4]
```diff
@@ -4,61 +4,26 @@
...
-  done()
-})
+- [onRequest](#onRequest)
+- [preParsing](#preParsing)
+- [preValidation](#preValidation)
+- [preHandler](#preHandler)
+- [preSerialization](#preSerialization)
+- [onError](#onError)
+- [onSend](#onSend)
+- [onResponse](#onResponse)
```
Comment (Eomm) [inline docs/Hooks.md]: I would improve this list also with the Application Hooks:

- Request/Reply Hooks
  - onRequest
  - etc...
- Application Hooks
  - onRegister
  - etc...

so the user will have a clear overview of all the hooks, that is not right now as you pointed out 😂

File: docs/Hooks.md (hunk 1 above)
Comment (RafaelGSS) [reply docs/Hooks.md]: for which?

File: docs/Hooks.md (hunk 2 above)
Comment (RafaelGSS) [reply docs/Hooks.md]: if the section name is changed too i agree

File: docs/Hooks.md (hunk 4 above)
Comment (RafaelGSS) [reply docs/Hooks.md]: Sure! I'll do make it today maybe.

Comment (RafaelGSS) [reply docs/Hooks.md]: @Eomm Changed! Can check the new version, please?

File: docs/Hooks.md [hunk 5]
```diff
@@ -4,61 +4,83 @@
...
-- `'preValidation'`
-- `'preHandler'`
-- `'preSerialization'`
-- `'onError'`
-- `'onSend'`
-- `'onResponse'`
-
-Example:
+
+- [Request/Reply Hooks](#request/reply-hooks)
```
Comment (Eomm) [inline docs/Hooks.md]: The link doesn't work due the `/`
```suggestion
- [Request/Reply Hooks](#requestreply-hooks)
```

File: docs/Hooks.md [hunk 6]
```diff
@@ -4,61 +4,83 @@
...
+
+## Request/Reply Hooks
+
+[Request](https://github.com/fastify/fastify/blob/master/docs/Request.md) and [Reply](https://github.com/fastify/fastify/blob/master/docs/Reply.md) are the core Fastify objects.<br/>
+`done` is the function to continue with the [lifecycle](https://github.com/fastify/fastify/blob/master/docs/Lifecycle.md).
+
+It is pretty easy to understand where each hook is executed by looking at the [lifecycle page](https://github.com/fastify/fastify/blob/master/docs/Lifecycle.md).<br>
+Hooks are affected by Fastify's encapsulation, and can thus be applied to selected routes. See the [Scopes](#scope) section for more information.
+
+### Manage Errors from a hook
```
Comment (Eomm) [inline docs/Hooks.md]: I think now the structure is near perfection 👏🏼

The `Manage Errors from a hook` and `Respond to a request from a hook` was ok at the bottom as before (after the [onRequest](https://github.com/fastify/fastify/blob/3456c5d2904cdfafb32d983995a3c3b3f3166d0d/docs/Hooks.md#onresponse) ). Sorry for the misunderstanding

Comment (RafaelGSS) [reply docs/Hooks.md]: Done

**Pull request merged**